### ISSUE TITLE
Create Table: url_type

### ISSUE THREAD

**Issue by fyliu** [hackforla/peopledepot#77]
### Overview

We need to create the url_type table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project.

#### Details

A table and a model are the same thing

### Action Items

- [x] identify and document table description (see spreadsheet under Resources)
  - [x] if not, reach out to PD leads
- [x] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [x] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
- [x] create a single model in Django (defining schema)
- [x] add seed data
- [x] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [x] write an API end point
- [x] write API unit tests
- [x] document the endpoint
- [ ] Release #41 from ice box

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

Type of the URL (ReadMe, Wiki, etc.)

### Data Fields

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] (PK) id - int - Record Id
   - [x] name - varchar
   - [x] description - textfield

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] project_url (one-to-many)

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

**Comment by fyliu** [comment]
This table's data has overlaps with some URL fields in the project table. We need to document what should go in the project table and what should go in project_url. It looks to me like the project_url is meant as a way to extend what's the project table without needing to change the db schema.

- [ ] create documentation issue to guide what urls should use project_url and which ones should be in project

- relates to #41

**Comment by joshuayhwu** [comment]
```
class URL_Type(AbstractBaseModel):
	"""
	user type
	"""
	id = models.IntegerField(primary_key=True)
	name = models.CharField(max_length=255)
	description = models.CharField(max_length=255)

	def __str__(self): 
		return f"{self.name}" 

```

**Comment by fyliu** [comment]
- I looked at the #78 and don't see anything that blocks this issue from being worked on.

**Comment by ExperimentsInHonesty** [comment]
@fang This issue needs to be cleaned up before it goes back into the prioritized backlog.  It should not have comments in the body.  it makes it too hard to follow

**Comment by fyliu** [comment]
Hidden all comments.

Turns out my last comment was saying that the previous comment has been resolved but I should have marked it hidden.

**Comment by shmonks** [comment]
Hi @dmartin4820 - please consider the following questions and add an update below:
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (if necessary): "Add any pictures that will help illustrate what you are working on."

**Comment by dmartin4820** [comment]
1. Progress: Just assigned a bit earlier this week. Not started yet.
2. Blockers: None yet.
3. Availability: Will be able to work on this weekend.
4. ETA: By next Tuesday assuming no blockers.
5. N/A

**Comment by dmartin4820** [comment]
- Created the PR but didn't check off: Release https://github.com/hackforla/peopledepot/issues/41 from ice box
Will complete and check off after the PR is merged.

**Comment by dmartin4820** [comment]
For weekly meeting today 

Progress: Got reviews the PR. I need to just implement the changes in the comments. 
Blockers: None yet
Availability: Flexible between 5pm - 9pm on weekdays.
ETA: Will work between tomorrow and over the weekend

**Comment by dmartin4820** [comment]
For 10/24/2024 meeting:
I won't be there.

Progress: Nothing this past week.
Blockers: Still in review, but I have other tasks to make progress on
Availability: Flexible between 5pm - 9pm on weekdays.
ETA: Will work over the weekend
